test/PushNotificationTests/APITests.cpp

```diff
@@ -0,0 +1,239 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include <TestDef.h>
+#include "Shared.h"
+#include "MockBackgroundTaskInstance.h"
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+
+using namespace winrt::Windows::ApplicationModel::Activation;
+using namespace winrt::Windows::ApplicationModel::Background;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::Management::Deployment;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::System;
+
+namespace Test::PushNotifications
+{
+    class APITests
+    {
+    private:
+        wil::unique_event m_failed;
+
+        const std::wstring c_testPackageFile = g_deploymentDir + L"PushNotificationsTestAppPackage.msix";
+        const std::wstring c_testPackageFullName = L"PushNotificationsTestAppPackage_1.0.0.0_" PROJECTREUNION_TEST_PACKAGE_DDLM_ARCHITECTURE L"__8wekyb3d8bbwe";
+        const std::wstring c_testVCLibsPackageFile = g_deploymentDir + L"VCLibs.appx";
+
+    public:
+        BEGIN_TEST_CLASS(APITests)
+            TEST_CLASS_PROPERTY(L"Description", L"Project Reunion Push Notifications test")
+            TEST_CLASS_PROPERTY(L"ThreadingModel", L"MTA")
+            TEST_CLASS_PROPERTY(L"RunAs:Class", L"RestrictedUser")
+        END_TEST_CLASS()
+
+        TEST_CLASS_SETUP(ClassInit)
+        {
+            try
+            {
+                InstallPackage(c_testVCLibsPackageFile);
+            }
+            catch (...)
```

Comment:
Do you need to catch each error separately?

Also, you should install PRfwk before PushTestPackage (register packages in order of layering/dependencies from the bottom/most-atomic on up) is best (and often, necessary).

Would simplify down to

```
TEST_CLASS_SETUP(ClassInit)
{
    try
    {
        TP::AddPackage_ProjectReunionFramework();
        InstallPackage(GetTestVCLibsPackageFile());
        InstallPackage(GetTestPackageFile());
    }
    catch (...)
    {
        return false;
    }
}
```

Reply:
If an exception is thrown, what TAEF display and log?
You may want some kind of logging or message before the return false.
Or perhaps catch nothing and let TAEF do the catching? (does it?)

Reply:
TAEF will display the error and block all the tests from running.